torch_xla/experimental/xla_sharding.py

```diff
@@ -319,8 +319,24 @@ def _get_sharding_type(partition_spec: Tuple[Union[int, None]],
   return sharding_type
 
 
-def _get_tile_assignment(mesh: Mesh) -> List[int]:
-  return mesh.get_logical_mesh().tolist()
+def _get_tile_assignment(mesh: Mesh,
+                         partition_spec: Tuple[Union[int, None]]) -> List[int]:
+  # Use Torch.tensor here to make use of the torch.transpose_
+  mesh_list_tensor = torch.tensor(mesh.get_logical_mesh().tolist())
+  # This is partial sharding case, tile_assigniment will be ignore in favor of
+  # group_assignment and replication_groups.
+  if (mesh_list_tensor.dim() != len(partition_spec)):
+    return mesh_list_tensor.tolist()
+  partition_spec_list = list(partition_spec)
+  for i in range(len(partition_spec_list)):
+    if partition_spec_list[i] == None:
```

Comment:
Not quite understanding this part and then it doesn't seem like you have a test case for it as well.

Reply:
For partial partial and replicated, the sharding spec will contains Nones. For example `[0, 1, None]`, `[None, None]`, we can't call permute in a array with `None`, so we have to replace them with numbers.

That being said, I am still working on partial partion on next pr, this check here just to make sure fully replicated and partial partion does not crash.

Reply:
without this, existing test for fully replicated will error out. since we can't do `tensor.permute([2, 1, None])`

Reply:
Thanks for the context. LGTM.